File: E_arrays6_intersection_arrays.cpp
//in progress

class Solution {
public:
        vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
               
        vector<int> nums3;
        
        int i,j;
        
        for (i=0; i<nums1.size();i++){

            for(j=0;j<nums2.size();j++){
                

                if (nums2[j]==nums1[i]){
                    
                    nums3.push_back(nums1[i]);
                    nums2.erase(nums2.begin()+j);
                    break;
                }
                
                
            }

        }
                      
        return nums3;
        
    }
};


//Other efficient runtime solution:
// class Solution {
// public:
//     vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        
//         map<int,int> table;
//         vector<int> ans;
        
//         for(int x:nums1) table[x]++;
//         for(int x:nums2)
//             if(table.count(x) && table[x]>0){
//                 table[x]--;
//                 ans.push_back(x);
//             }
        
//         return ans;
        
        
//     }
// };

File: E_arrays6_intersection_arrays.py
# Given two integer arrays nums1 and nums2, return an array of their intersection. 
# Each element in the result must appear as many times as it shows in both arrays and you may return the result in any order.


#More efficient, using break
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
           
        nums3=[]
        
        for item in nums1:

            for item2 in nums2:

                if item2==item:
                    
                    nums3.append(item)
                    nums2.remove(item2)
                    #we exit from the second for to avoid comparing item again
                    break
               
        return nums3



#Less efficient
class Solution2:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        
        
        nums3=[]
        repeat_found=False
        
        for item in nums1:

            for item2 in nums2:

                if (item2==item and repeat_found==False):

                    repeat_found=True
                    nums3.append(item)
                    nums2.remove(item2)
            
            repeat_found=False

            
        return nums3
